k8s/k8s.go

```diff
@@ -2069,6 +2073,29 @@ func (k *k8sOps) IsPodBeingManaged(pod v1.Pod) bool {
 	return false
 }
 
+func (k *k8sOps) ValidatePod(pod *v1.Pod) error {
+	t := func() (interface{}, bool, error) {
+		logrus.Infof("RK=> Pod Name: %s, Pod UID %s, Pod Status: %s", pod.Name, pod.UID, pod.Status.Phase)
+		currPod, err := k.GetPodByUID(pod.UID, pod.Namespace)
+		//currPod, err := k.GetPodByName(pod.Name, pod.Namespace)
+		logrus.Infof("RK=> Curr: Pod Name: %s, Pod UID %s, Pod Status: %s", pod.Name, pod.UID, pod.Status.Phase)
+		if err != nil {
+			return "", true, fmt.Errorf("Could not get Pod [%s] %s", pod.Namespace, pod.Name)
+		}
+
+		ready := k.IsPodReady(*currPod)
```

Comment:
ideally IsPodReady should have taken a pointer instead of the whole object, but we can do this change later.

Reply:
Ok. will add a TODO.